### PR TITLE
feat: add cron job to migrate team/org billing data from bookings

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

Adds a cron job to incrementally migrate team and organization billing data from the `Team.metadata` JSON field to the dedicated `TeamBilling` and `OrganizationBilling` tables.

**Migration Strategy:**
- Queries bookings from the last 24 hours (configurable via `lookbackHours` parameter)
- Extracts unique team IDs from those bookings
- Migrates billing data for teams that haven't been migrated yet
- Skips teams that already have billing records or lack subscription data

This approach has zero impact on the booking flow and processes teams incrementally as they become active through new bookings.

**Link to Devin run:** https://app.devin.ai/sessions/31059a89a9ed46e7a3b20fb9531b2166
**Requested by:** joe@cal.com (@joeauyeung)

## Updates Since Last Revision

Refactored to follow Cal.com's architectural patterns per reviewer feedback:
- **Thin controller pattern**: Route handler now only handles HTTP concerns (auth, request parsing, response formatting)
- **Repository pattern**: Service uses `BookingRepository` and `ITeamBillingDataRepository` instead of calling Prisma directly
- **Dependency injection**: All repositories obtained via DI container (`getTeamBillingDataRepository()`, `getBillingRepositoryFactory()`)
- **Factory pattern**: `BillingRepositoryFactory` injected to create appropriate billing repository (team vs organization) based on entity type
- **Unit tests added**: 13 comprehensive tests for `BillingMigrationService` covering migration scenarios, skip conditions, error handling, and edge cases
- Merged with main branch and adapted to new team billing repository structure at `packages/features/ee/billing/repository/teamBillingData/`

## Key Implementation Details

**Architecture:**
- `route.ts` - Thin controller handling auth and HTTP concerns only
- `BillingMigrationService` - Business logic for migration orchestration, receives dependencies via constructor
- `BookingRepository.findDistinctTeamIdsByCreatedDateRange()` - New method to find teams from recent bookings
- `PrismaTeamBillingDataRepository.findByIdIncludeBillingRecords()` - New method to fetch team with billing status
- `Billing.ts` container - Added `getBillingRepositoryFactory()` export for DI

**Data Mapping:**
| Metadata Field | Billing Table Field |
|---------------|-------------------|
| `subscriptionId` | `subscriptionId` |
| `subscriptionItemId` | `subscriptionItemId` |
| `paymentId` | `customerId` ⚠️ **Needs verification** |
| - | `status` (default: "ACTIVE") ⚠️ **May need refinement** |
| `isOrganization` | `planName` (TEAM or ORGANIZATION) |

**Authentication:** Uses `CRON_API_KEY` or `CRON_SECRET` (consistent with other cron jobs)

## Important Review Points

⚠️ **Critical items to verify:**

1. **Metadata field mapping** - Confirm that `paymentId` in metadata is actually the Stripe `customerId`
2. **Default status** - All migrated teams get status "ACTIVE" - is this correct for teams in trial, past_due, etc.?
3. **Repository methods** - Verify `findDistinctTeamIdsByCreatedDateRange` and `findByIdIncludeBillingRecords` implementations are correct
4. **DI container** - Verify `billingRepositoryFactoryModuleLoader` loads correctly with its dependencies

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - This is a new internal cron job, no user-facing documentation needed. README included in PR.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

**Prerequisites:**
- Set `CRON_API_KEY` or `CRON_SECRET` in environment variables
- Have teams with subscription data in `Team.metadata`
- Have recent bookings for those teams

**Manual Testing:**
```bash
# Test with default 24-hour lookback
curl -X POST "http://localhost:3000/api/cron/migrate-billing" \
  -H "Authorization: Bearer ${CRON_SECRET}"

# Test with custom lookback period
curl -X POST "http://localhost:3000/api/cron/migrate-billing?lookbackHours=48" \
  -H "Authorization: Bearer ${CRON_SECRET}"
```

**Verify:**
1. Check response shows correct counts (teamsFound, migrated, skipped, errors)
2. Query `TeamBilling` and `OrganizationBilling` tables to verify data was inserted
3. Re-run the job to confirm already-migrated teams are skipped

**Unit Tests:**
```bash
yarn test packages/features/ee/billing/services/BillingMigrationService.test.ts
```

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have checked if my changes generate no new warnings
- [x] Added comprehensive README documentation

### CODE REVIEW

File: apps/web/app/api/cron/migrate-billing/route.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
+import { defaultResponderForAppDir } from "app/api/defaultResponderForAppDir";
```
Comment (keithwillcode) [inline apps/web/app/api/cron/migrate-billing/route.ts:1]: This logic should not live directly in this route.ts. It goes against our patterns of having these entry points be thin. Let’s move logic to application services.

**Review by keithwillcode — CHANGES_REQUESTED** [review]
